device/iot-device-samples/android-sample/app/proguard-rules.pro

```diff
@@ -15,3 +15,26 @@
 #-keepclassmembers class fqcn.of.javascript.interface.for.webview {
 #   public *;
 #}
+
+# JSON parser classes don't mix well with minification because JSON parser classes use reflection
+# based on the name of the field to create the JSON, and minification changes those names. As a result
+# all the message payload serializing and deserializing breaks. These lines explicitly exclude
+# the packages where we keep our JSON parsing classes from minification to avoid this issue
+-keep class com.microsoft.azure.sdk.iot.deps.serializer.* { *; }
+-keep class com.microsoft.azure.sdk.iot.deps.twin.* { *; }
+-keep class com.microsoft.azure.sdk.iot.device.edge.MethodRequest { *; }
+-keep class com.microsoft.azure.sdk.iot.device.hsm.parser.ErrorResponse { *; }
+-keep class com.microsoft.azure.sdk.iot.device.hsm.parser.SignRequest { *; }
+
+# These are the recommended exclusions when using the IoT Hub service client SDK. They are commented
+# out because this sample only uses the IoT Hub device client SDK
+#-keep class com.microsoft.azure.sdk.iot.service.ImportMode { *; }
```

Comment:
This is very useful info, but is the device client sample the right place for this? Do we have service client/ provisioning etc android samples where we could place these lines instead?

Reply:
We don't have an android sample for service client or for our provisioning clients, so this is the best place for now